src/stemmablenote.ts

```diff
@@ -189,14 +202,12 @@ export abstract class StemmableNote extends Note {
 
     // Use stemBeamExtension with beams
     if (this.beam) {
-      return glyphProps.stemBeamExtension;
-    }
-
-    if (glyphProps) {
-      return this.getStemDirection() === Stem.UP ? glyphProps.stemUpExtension : glyphProps.stemDownExtension;
+      return glyphProps.stemBeamExtension * this.getStaveNoteScale();
     }
 
-    return 0;
+    return this.flag.getHeight() > Stem.HEIGHT * this.getStaveNoteScale()
```

Comment:
Can we add a comment here? I don't understand this line.

Maybe consider extracting two local vars for:
`this.flag.getHeight()` and 
`Stem.HEIGHT * this.getStaveNoteScale()`

Reply:
done